### ISSUE TITLE
R506 false positive when raise is used in elif

### ISSUE THREAD

**Issue by nijel** [Afonasev/flake8-return#130]
* Date you used flake8-return: 15th December 2022
* flake8-return version used, if any: 1.2.0
* Python version, if any: 3.9.2
* Operating System: Linux

### Description

Linting following code:

```py
def test_2(bar):
    if bar == 3:
        baz = 1
    elif bar == 4:
        raise Exception()
    else:
        baz = 2
    return baz
```

### What I Did

```console
$ flake8 test.py 
test.py:4:5: R506 unnecessary else after raise statement.
```

The else is necessary as there is additional if before which has side effects and removing else would overwrite that. The code is simplified from real-world code.

**Comment by nijel** [comment]
The very same issue is with return and R505:

```py
def test_2(bar):
    if bar == 3:
        baz = 1
    elif bar == 4:
        return 3
    else:
        baz = 2
    return baz
```

```console
$ flake8 test.py 
test.py:4:5: R505 unnecessary else after return statement.
```
